docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+
+### 3.5 Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
+_Figure 8: Command Class Diagram_
+
+All Commands contain a command word constant named as `COMMAND_*` (as underlined in _Figure 8_),\
+e.g. `protected static final String COMMAND_XYZ = "xyz";`\
+These constants are used by the `CommandHandler` to map to each `Command`.
+
+#### Description
+The `Command` component contains the `abstract Command` class and its extensions. The extensions 
+of `Command` are the `AddCommand`, `CreditScoreCommand`, `ViewCommand`, etc...
+
+#### Design
+The only `abstract` method of `Command` is `execute(...)` where it is called everytime a `Command`
+object is created. Most of the input validation is done in the constructor of each Command object. 
+1. Firstly, arguments are checked for validity (if any):
+   1. Valid options: options only for each `Command`. E.g. `-a` for the amount `add` is valid.
+   2. No duplicate options: options are not repeated. E.g. `-a 200 -a 200` or `-l -l` is invalid.
+   3. No conflict options: mutually exclusive options, options that cannot be input at the same time.
+      E.g. ViewCommand implements `{-e | -l | -s}` options, `view -s -l` is considered a conflict of options.
+2. Secondly, option values are validated (if any):
+   * E.g. The input `return -i 2 -d 20122012` has option values of `"2"` for `-i` and `"20122012"` for `-d`.
+     Each option value is validated based on the input validation methods for each data type.
+
+If no violations are present in the arguments, then the subsequent `Command` object is returned.\
+If violations occur at any point of the input validation, the `Command` is not created and `CommandException` 
+is thrown back to the `CommandHandler`.
+
+### 3.6 RecordList Component
+The `recordlist` class maintains an internal arraylist of record objects used throughout Finux's execution.
+
+
+### 3.7 Storage Component
+...
+
+## 4. Implementation
+
+---
+
+This section introduces the specific implementation details and design thought processes
+of some features in **Finux**.
+
+### 4.1 Add Feature
+The `add` feature aims to allow users to add *expense*, *loan*, and *saving* records.
+When adding an expense `add -e bread loaf -a 2.50 -d today`, or\
+adding a savings `add -s week's savings -a 100 -d 28/03/2021`, or\
+adding a loan `add -l loan to gerard -a 200 -d 12012021 -p Gerard`,
+the `ParserHandler` will parse the input for `CommandHandler` to create the `AddCommand` object.
+By calling the `execute()` method, the respective `Expense`, `Saving` or `Loan` object is added 
+into the `RecordList`.
+
+#### 4.1.1 Current Implementation
+...
+
+#### 4.3.2 Design Consideration
+...
+
+### 4.2 List Feature
+The `list` feature allows Finux users to list records that they have entered into the system.
+
+To list loans:
+
+`list -l`
+
+To list expenses:
+
+`list -e`
+
+To list savings:
+
+`list -s`
```

Comment:
For this part, perhaps you might want to remove it if you have nothing to share about its implementation.

Reply:
I think that was some sample placeholder for that section, we will renew it. Thanks!